packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.types.ts

```diff
@@ -636,8 +636,8 @@ export interface IDetailsListStyles {
 }
 
 export interface IDetailsGroupRenderProps extends IGroupRenderProps {
-  onRenderFooter?: IRenderFunction<IDetailsGroupDividerProps>;
-  onRenderHeader?: IRenderFunction<IDetailsGroupDividerProps>;
+  onRenderFooter?: IRenderFunction<IGroupFooterProps & IDetailsGroupDividerProps>;
```

Comment:
I'm unsure about these changes. I thought this might at least have to be `|` for backwards-compatibility. Am I incorrect?

Reply:
Using `&` is requiring *both* sets of props to be provided at all times. Unless that's the intention here, this should be `|` instead.

Another question I have is: How do render function implementers know which set of props they've been provided?

Reply:
@Vitalius1, is this type change related to being able to pass the `IGroupFooterStyleProps` in the `DetailsList` grouped case? i.e.

```tsx
  /**
   * Style function to be passed in to override the themed or default styles
   */
  styles?: IStyleFunctionOrObject<IGroupFooterStyleProps, IGroupFooterStyles>;
```

I made the changes locally to better understand these changes and that is the limitation I ran into. The one "_positive_" is that _all_ fields on `IGroupFooterProps`, `IGroupDividerProps`, and `IDetailsItemProps` are _optional_...

Reply:
I think it would be safer for consumers to keep the changes scoped to `GroupedList` as done here:

https://github.com/OfficeDev/office-ui-fabric-react/compare/master...KevinTCoughlin:keco/grouped-list-types

In both branches `DetailsList` group divider render functions do not utilize the `styles` which the type intersection above is providing. So that change seems unnecessary _for now_. Happy to be corrected.

Reply:
@JasonGore @KevinTCoughlin 
Thanks to pointing out to this change. I believe I just was going with the wave of cleaning up that `IGroupDividerProps` usage everywhere and didn't realize it's a custom renderer of the `DetailsList`. Reversed the change and scoped the changes to only `GroupedList`